internal/backend/basicstation/structs/radio_meta_data.go

```diff
@@ -83,6 +85,15 @@ func SetRadioMetaDataToProto(loraBand band.Band, gatewayID lorawan.EUI64, rmd Ra
 			return errors.Wrap(err, "timestamp proto error")
 		}
 
+	} else {
```

Comment:
Given that both the `rxtime` and `gpstime` are provided, I think it is better that both are handled individually, such that:

* `GPSTime` gets mapped to `TimeSinceGpsEpoch` and
* `RxTime` gets mapped to `Time`

The above conversion (`gpsTimeTime := time.Time(gps.NewTimeFromTimeSinceGPSEpoch(gpsTimeDur))`) was there since I wasn't aware that also the `rxtime` was provided (or it was introduced in a later version) and I wanted to set the `Time` field.

Thus:

```go
if gpsTime := rmd.UpInfo.GPSTime; gpsTime != 0 {

}

if rxTime := rmd.UpInfo.RxTime; rxTime != 0 {

}
```

Reply:
@brocaar 
Yes, it's better to handle individually
I'II make changes and update this PR

Thanks